pl_examples/quantization/imagenet.py

```diff
@@ -0,0 +1,374 @@
+        )
+        parser.add_argument(
+            "-b",
+            "--batch-size",
+            default=256,
+            type=int,
+            metavar="N",
+            help="mini-batch size (default: 256), this is the total batch size of all GPUs on the current node"
+            " when using Data Parallel or Distributed Data Parallel",
+        )
+        parser.add_argument(
+            "--lr", "--learning-rate", default=0.001, type=float, metavar="LR", help="initial learning rate", dest="lr"
+        )
+        parser.add_argument("--momentum", default=0.9, type=float, metavar="M", help="momentum")
+        parser.add_argument(
+            "--wd",
+            "--weight-decay",
+            default=1e-4,
+            type=float,
+            metavar="W",
+            help="weight decay (default: 1e-4)",
+            dest="weight_decay",
+        )
+        parser.add_argument("--pretrained", dest="pretrained", action="store_true", help="use pre-trained model")
+        return parent_parser
+
+
+def main(args: Namespace) -> None:
+    if args.seed is not None:
+        pl.seed_everything(args.seed)
+
+    print(args)
+
+    if args.accelerator == "ddp":
+        # When using a single GPU per process and per
+        # DistributedDataParallel, we need to divide the batch size
+        # ourselves based on the total number of GPUs we have
+        args.batch_size = int(args.batch_size / max(1, args.gpus))
+        args.workers = int(args.workers / max(1, args.gpus))
+
+    model = ImageNetLightningModel(**vars(args))
+    if args.evaluate:
+        if args.precision_mode == 'int8':
+            model = pl.compressor.intel.neural_compressor.load(model, args.checkpoint)
+        dataloader = model.val_dataloader()
+        validate(dataloader, model, args)
+        return
+
+    quantizer = pl.IncQuantization('config/quantization.yaml', model, calib_dataloader=model.val_dataloader(),
+                                   metric_name="val_acc1")
+    if quantizer.approach == pl.QuantizationMode.QAT.value:
+        from pytorch_lightning.callbacks.early_stopping import EarlyStopping
+        early_stop_callback = EarlyStopping(monitor="val_acc1", min_delta=0.00, patience=2, verbose=False, mode="max")
+        trainer = pl.Trainer.from_argparse_args(args, callbacks=[early_stop_callback])
+    else:
+        trainer = pl.Trainer.from_argparse_args(args)
+    quantizer.trainer = trainer
+    if quantizer.approach == pl.QuantizationMode.PTQ_DYNAMIC.value:
+        q_model = quantizer.fit_dynamic()
+    elif quantizer.approach == pl.QuantizationMode.PTQ_STATIC.value:
+        q_model = quantizer.fit_static()
+    elif quantizer.approach == pl.QuantizationMode.QAT.value:
+        q_model = quantizer.fit_aware_training()
+    else:
+        raise ValueError(f"Unknown quantization approach.")
+
+    quantizer.save(args.checkpoint)
+
+
+def validate(val_loader, model, args):
```

Comment:
This counter Lightning philosophy. You should use Trainer.validate to gather the metric and let's use TorchMetric.

Reply:
removed